### ISSUE TITLE
Can't start server assembled with maven

### ISSUE THREAD

**Issue by PurpleCrowbar** [magefree/mage#8856]
On Windows 10. Ran `mvn clean install -DskipTests` on root directory, then `mvn assembly:assembly` in Mage.Client and Mage.Server. I can run the generated startClient.bat and get to the client but when I try running the startServer.bat I get the following:

`INFO  2022-04-18 11:57:28,127 Starting MAGE server version 1.4.50-V2 (build: 2022-04-17 18:56)                           =>[main] Main.main
INFO  2022-04-18 11:57:28,129 Logging level: INFO                                                                        =>[main] Main.main
INFO  2022-04-18 11:57:28,129 Default charset: UTF-8                                                                     =>[main] Main.main
INFO  2022-04-18 11:57:28,132 Reading configuration from path=config\config.xml                                          =>[main] Main.main
Exception in thread "main" java.lang.NoClassDefFoundError: javax/xml/bind/JAXBContext
        at mage.server.util.ConfigFactory.loadFromFile(ConfigFactory.java:13)
        at mage.server.Main.main(Main.java:104)
Caused by: java.lang.ClassNotFoundException: javax.xml.bind.JAXBContext
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
        ... 2 more`

The issue is not that the versions are different as my problem isn't related to connecting to the server with a different version of the client, just that I can't start the server in the first place.

**Comment by JayDi85** [comment]
Bat-script uses default java version from your system, so you can run it with java 9 instead java 8 (you can run `cmd` -> `java --version` to check it).

If you want to run server from the sources then I recommends to run it as application (see wiki for [IntelliJ config example](https://github.com/magefree/mage/wiki/Setting-up-your-Development-Environment#running-server-and-client)). If you want to run it under java 8 then change bat-script with new path to `java.exe`.

![shot_220418_164934](https://user-images.githubusercontent.com/8344157/163811774-cacd7e52-95d9-4e86-9c72-fc7fef0d6768.png)

**Comment by PurpleCrowbar** [comment]
Thank you, I was able to get the server running by switching my active version of Java and then re-running the bat files. I set up IntelliJ Idea but can't seem to run the client or server as I get the following error when trying to do either:
`..\mage\Mage\src\main\java\mage\game\match\Match.java:13:37`
`java: package mage.game.result.ResultProtos does not exist`
I'm fine with using a different IDE and just running the bats to start the client / server though. Is there a way to set test mode to true for the server in the bat or do I have to edit a config or something?

**Comment by JayDi85** [comment]
1. "Result proto miss" error - it's auto-generated sources, so you must run
full build (or maven's install command).

2. Look at wiki about test mode and command line params:
https://github.com/magefree/mage/wiki/Development-Testing-Tools

**Comment by PurpleCrowbar** [comment]
In IntelliJ Idea I have set up the run configurations as specified on the ["Setting up your Development Environment" page](https://github.com/magefree/mage/wiki/Setting-up-your-Development-Environment#IntelliJ-Run-Configurations) and can mvn install and build the client and server from within IntelliJ just fine, but when I try to run the client or server with the configuration described on that page I just see `Writing classes... [mage-sets]` infinitely. I've already ran `mvn clean install -DskipTests` in the mage root, Mage.Client root, and Mage.Server root. Image below shows non-functional "Run Test Server" application:

![run test server](https://user-images.githubusercontent.com/26198472/164283818-6470d12b-f9fe-4a06-80c5-38361a3b2f7c.png)

I've tried building the client and server individually with `mvn clean install -DskipTests` each but couldn't run them from the IDE, and `mvn assembly:assembly` works for making zips (creates zip files in `mage/Mage.Client/target` and `mage/Mage.Server/target` respectively) which can be extracted and the .bat files ran, but the "Run Server", "Run Test Server", and "Run Client" within IntelliJ all don't run anything, and running `startServer.bat -testMode=true` from the command line just runs a regular server, not a test mode one, so not sure if it's possible to start a test mode server using just a bat file.
